edge/pkg/clusterd/mission_deployer.go

```diff
@@ -0,0 +1,348 @@
+			deployContentCmd := fmt.Sprintf("printf \"%s\" | %s delete --kubeconfig=%s -f - ", mission.Spec.Content, config.Config.KubectlCli, config.Config.Kubeconfig)
+			_, err := util.ExecCommandLine(deployContentCmd)
+			if err != nil {
+				klog.Errorf("Failed to revert the content of mission %v: %v", mission.Name, err)
+			} else {
+				klog.Infof("The content of mission %v is reverted.", mission.Name)
+			}
+		}
+	}
+
+	deleteMissionCmd := fmt.Sprintf("%s delete mission %s --kubeconfig=%s", config.Config.KubectlCli, mission.Name, config.Config.Kubeconfig)
+	if _, err := util.ExecCommandLine(deleteMissionCmd); err != nil {
+		return fmt.Errorf("Failed to delete mission %v: %v", mission.Name, err)
+	}
+
+	klog.Infof("Mission %v deleted successfully ", mission.Name)
+
+	return nil
+}
+
+func (m *MissionDeployer) DeleteMissionByName(name string) error {
+	mission, err := helper.GetMissionByName(name)
+	if err != nil {
+		return err
+	}
+
+	return m.DeleteMission(mission)
+}
+
+func (m *MissionDeployer) isMatchingMission(mission *edgeclustersv1.Mission) bool {
+	// if the placement field is empty, it matches all the edge clusters
+	if len(mission.Spec.Placement.Clusters) == 0 && len(mission.Spec.Placement.MatchLabels) == 0 {
+		return true
+	}
+
+	for _, matchingCluster := range mission.Spec.Placement.Clusters {
+		if config.Config.Name == matchingCluster.Name {
+			return true
+		}
+	}
+
+	// TODO: use k8s Labels operator to match
+	if len(mission.Spec.Placement.MatchLabels) == 0 {
+		return false
+	}
+
+	for k, v := range mission.Spec.Placement.MatchLabels {
+		if val, ok := config.Config.Labels[k]; ok && val == v {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (m *MissionDeployer) AlignMissionList(missionList []*edgeclustersv1.Mission) error {
+	missionMap := map[string]bool{}
+	var errs []error
+	for _, mi := range missionList {
+		missionMap[mi.Name] = true
+		if err := m.ApplyMission(mi); err != nil {
+			// Try to apply as many missions as possible, so move on after hitting error
+			errs = append(errs, fmt.Errorf("Error when applying mission %s: %v", mi.Name, err))
+		}
+	}
+
+	localMissions := helper.GetLocalClusterScopeResourceNames("missions", "")
+
+	for _, mi := range localMissions {
+		if _, exists := missionMap[mi]; !exists {
+			if err := m.DeleteMissionByName(mi); err != nil {
+				// Try to remove as many missions as possible, so move on after hitting error
+				errs = append(errs, fmt.Errorf("Error when deleting mission %s: %v", mi, err))
+			}
+		}
+	}
+
+	if len(errs) == 0 {
+		return nil
+	}
+
+	return fmt.Errorf("Hit the errors in mission align: %v", errs)
+}
+
+// create a yaml to use by "kubectl apply" command
+func buildMissionYaml(input *edgeclustersv1.Mission) (string, error) {
+	// probably due to the json encoder in arktos, the commmnd "kubectl apply missiong" in arktos
+	// fails if the mission.StateCheck.Command is nil or empty.
+	// We trick it with a string with one space.
+	// TODO: We need a non-hacky way.
```

Comment:
create a new issue for this.

Reply:
After a second thought, I believe the current trick is acceptable. I removed this "TODO" comment.